### PR TITLE
Backport animation to 10.6.11

### PR DESCRIPTION
**Pull request by ryantmer**
This backports #323 and #329 into the `v10.6.11` branch for OnPrem releases. (Just cherry-picked the commits from those PRs and resolved conflicts, nothing fancy.)

TODO:

- [x] Sanity check

### CODE REVIEW

File: d2l-all-courses.html [hunk 1]
```diff
@@ -437,15 +420,9 @@ <h2 class="d2l-heading-3">{{localize('unpinnedCourses')}}</h2>
...
-					this.filteredPinnedEnrollments.length > 0 ||
-					this.filteredPinnedEnrollmentsQueue.length > 0;
-				// Need to factor in queues here - if user has 1 enrollment, it will briefly
-				// disappear from both enrollments arrays when pinned/unpinned
-				this._hasFilteredEnrollments =
-					this._hasFilteredPinnedEnrollments ||
-					this.filteredUnpinnedEnrollments.length > 0 ||
-					this.filteredUnpinnedEnrollmentsQueue.length > 0;
+				this._hasFilteredPinnedEnrollments = this.filteredPinnedEnrollments.length > 0;
+				this._hasFilteredUnpinnedEnrollments = this.filteredUnpinnedEnrollments.length > 0;
```
Comment (omsmith) [inline d2l-all-courses.html]: This prop isn't a thing

Comment (ryantmer) [inline d2l-all-courses.html]: `filteredUnpinnedEnrollments` you mean?

Comment (omsmith) [inline d2l-all-courses.html]: _hasFilteredUnpinnedEnrollments. It was added later.

File: d2l-all-courses.html [hunk 2]
```diff
@@ -350,11 +324,6 @@ <h2 class="d2l-heading-3">{{localize('unpinnedCourses')}}</h2>
 					value: false
 				},
 
-				// True when there are any filtered enrollments (pinned or unpinned)
```
Comment (omsmith) [inline d2l-all-courses.html]: Where did this go?

Comment (ryantmer) [inline d2l-all-courses.html]: The property was defined and assigned to, but not used. Not sure if it was replaced by another thing, or...?

Comment (omsmith) [inline d2l-all-courses.html]: But you're changing more things than necessary now :P

Comment (omsmith) [inline d2l-all-courses.html]: i.e. seems that should maybe be fixed in master, but not here

Comment (ryantmer) [inline d2l-all-courses.html]: Yeah fair point!

**Comment by ryantmer** [comment]
Played with this locally - appears to work fine with pinning/unpinning and filtering/sorting courses.

**Pull request merged**